backend/btrixcloud/storages.py

```diff
@@ -24,462 +25,497 @@
     sync_get_log_stream,
 )
 
+from .utils import is_bool
+
 
 CHUNK_SIZE = 1024 * 256
 
 
 # ============================================================================
-def init_storages_api(org_ops, crawl_manager, user_dep):
-    """API for updating storage for an org"""
+class StorageOps:
+    """All storage handling, download/upload operations"""
 
-    router = org_ops.router
-    org_owner_dep = org_ops.org_owner_dep
+    def __init__(self, org_ops, crawl_manager):
+        self.storages = {}
+
+        self.org_ops = org_ops
+        self.crawl_manager = crawl_manager
+
+        self.is_local_minio = is_bool(os.environ.get("IS_LOCAL_MINIO"))
+
+        with open("/tmp/storages/storages.json", encoding="utf-8") as fh:
+            storage_list = json.loads(fh.read())
+
+        for storage in storage_list:
+            name = storage.get("name")
+            type_ = storage.get("type", "s3")
+            if type_ == "s3":
+                self.storages[name] = self._create_s3_storage(storage)
+            else:
+                # expand when additional storage options are supported
+                raise TypeError("Only s3 storage supported for now")
+
+    def _create_s3_storage(self, storage):
+        """create S3Storage object"""
+        endpoint_url = storage["endpoint_url"]
+        bucket_name = storage.get("bucket_name")
+        if bucket_name:
+            endpoint_url += bucket_name + "/"
+
+        if self.is_local_minio:
+            access_endpoint_url = "/data/"
+            use_access_for_presign = False
+        else:
+            access_endpoint_url = storage.get("access_endpoint_url") or endpoint_url
+            use_access_for_presign = True
+
+        return S3Storage(
+            access_key=storage["access_key"],
+            secret_key=storage["secret_key"],
+            region=storage.get("region", ""),
+            endpoint_url=endpoint_url,
+            access_endpoint_url=access_endpoint_url,
+            use_access_for_presign=use_access_for_presign,
+        )
+
+    def has_storage(self, name):
+        """assert the specified storage exists"""
+        return name in self.storages
 
-    # pylint: disable=bare-except, raise-missing-from
-    @router.patch("/storage", tags=["organizations"])
     async def update_storage(
-        storage: Union[S3Storage, DefaultStorage],
-        org: Organization = Depends(org_owner_dep),
-        user: User = Depends(user_dep),
+        self, storage: Union[S3Storage, DefaultStorage], org: Organization, user: User
     ):
+        """update storage for org"""
         if storage.type == "default":
```

Comment:
Are we actually supporting non-s3 storage at this point? Should we remove this option and just require s3 for now? Even locally we're using minio. Might help consolidate storage if we always assume a consistent type (at least for now), rather than having a bunch of `if/elses` for something we don't seem to be using.